File: Educational_51_B.cpp
// code_report Solution
// https://youtu.be/vcF8mTjgt6o

#include <iostream>

using namespace std;
using ll = long long;

int main () {
   ll l, r;
   cin >> l >> r;
   cout << "YES" << endl; 
   for (ll i = l; i < r + 1; i += 2)
      cout << i << ' ' << i + 1 << endl;
   return 0;
}

File: Educational_51_B.java
// code_report Solution
// https://youtu.be/vcF8mTjgt6o

import java.util.*;
import java.io.*;

public class Main {
     public static void main(String []args) {
        Scanner sc = new Scanner(System.in);
        Long l = sc.nextLong();
        Long r = sc.nextLong();
        System.out.println("YES");
        for (Long i = l; i < r; i += 2)
            System.out.println(Long.toString(i) + ' ' + 
                               Long.toString(i+1));
     }
}
